roles/jd-server/src/lib/job_declarator/mod.rs

```diff
@@ -90,113 +100,94 @@ impl JobDeclaratorDownstream {
-                TransactionState::Present(_) => continue,
-                TransactionState::ToBeRetrievedFromNodeMempool(tx) => {
-                    transactions_to_be_retrieved.push(tx)
-                }
-                TransactionState::Missing => continue,
-            }
-        }
-        let mempool_ = self_mutex
-            .safe_lock(|a| a.mempool.clone())
-            .map_err(|e| JdsError::PoisonLock(e.to_string()))?;
-        let client = mempool_
-            .clone()
-            .safe_lock(|a| a.get_client())
-            .map_err(|e| JdsError::PoisonLock(e.to_string()))?
-            .ok_or(JdsError::MempoolError(
-                mempool::error::JdsMempoolError::NoClient,
-            ))?;
-        for txid in transactions_to_be_retrieved {
-            let transaction = client
-                .get_raw_transaction(&txid.to_string(), None)
-                .await
-                .map_err(|e| JdsError::MempoolError(mempool::error::JdsMempoolError::Rpc(e)))?;
-            let txid = transaction.txid();
-            mempool::JDsMempool::add_tx_data_to_mempool(
-                mempool_.clone(),
-                txid,
-                Some(transaction.clone()),
-            );
-            new_transactions.push(transaction);
-        }
-        for transaction in new_transactions {
-            self_mutex
-                .clone()
-                .safe_lock(|a| {
-                    for transaction_with_state in &mut a.declared_mining_job.1 {
-                        match transaction_with_state {
-                            TransactionState::Present(_) => continue,
-                            TransactionState::ToBeRetrievedFromNodeMempool(_) => {
-                                *transaction_with_state =
-                                    TransactionState::Present(transaction.txid());
-                                break;
-                            }
-                            TransactionState::Missing => continue,
-                        }
-                    }
-                })
-                .map_err(|e| JdsError::PoisonLock(e.to_string()))?
-        }
-        Ok(())
-    }
-
     fn get_block_hex(
         self_mutex: Arc<Mutex<Self>>,
         message: SubmitSolutionJd,
     ) -> Result<String, Box<JdsError>> {
-        let (last_declare_, transactions_with_state, _) = self_mutex
+        let (last_declare_, _, _) = self_mutex
+            .clone()
+            .safe_lock(|x| x.declared_mining_job.clone())
+            .map_err(|e| Box::new(JdsError::PoisonLock(e.to_string())))?;
+        let last_declare = last_declare_.ok_or(Box::new(JdsError::NoLastDeclaredJob))?;
+        let transactions_list = Self::are_all_job_transactions_present(self_mutex)?;
+        let block: Block =
+            roles_logic_sv2::utils::BlockCreator::new(last_declare, transactions_list, message)
+                .into();
+        Ok(hex::encode(serialize(&block)))
+    }
+
+    fn are_all_job_transactions_present(
```

Comment:
change the function name is something else, this seems just a condition check but it do something